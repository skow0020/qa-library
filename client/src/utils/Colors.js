const colors = {
  red: '#f44336',
  blue: '#2196f3',
  green: '#4caf50',
  purple: '#9c27b0',
  black: '#000000',
  grey: '#3B3D40',
  teal: '#009688',
  cyan: '#00bcd4',
  white: '#FFFFFF',


  // Standard Colors
  darkGrey: '#3B3D40',
  midGrey1: '#81858D',
  midGrey3: '#5A5D64',
  midGrey2: '#6D7078',
  lightGrey1: '#F2F3F4',
  lightGrey2: '#DDE1E3',
  lightGrey3: '#CFD3D6',

  // Buttons
  primary: '#006f51',
  primaryHover: '#00A881',
  secondary: '#000000',
  secondaryHover: '#383838',

  // Info Colors
  success: '#4B8400',
  error: '#E31F2D',
  warning: '#FFD364',
  info: '#CFD3D6',

  // Active/Deactive
  active: '#4B8400',
  deactivated: '#6D7078',

  // Style
  aluminium: '#A8ABB1',

  // Tooltips
  tooltipBackground: '#1B1B1D',
  appBarBackground: '#005B46'
};

export default colors;