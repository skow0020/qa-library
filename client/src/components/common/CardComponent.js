import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import CardHeader from '@material-ui/core/CardHeader';
import Colors from 'utils/Colors';
import PropTypes from 'prop-types';
import React from 'react';
import Typography from '@material-ui/core/Typography';
import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles(() => ({
  cardImage: {
    position: 'relative',
    minHeight: '10.3125rem',
    backgroundSize: 'cover',
    backgroundPosition: 'center',
    backgroundRepeat: 'no-repeat'
  }
}));

export default function ComponentCard(props) {
  const classes = useStyles();
  const { idx, url, title, subheader, avatar, backgroundImage, body, urlTarget, children } = props;

  return (
    <Card id={idx} className="card-post">
      <a href={url} target={urlTarget} rel="noopener noreferrer" aria-label="Navigate to the url" style={{ color: Colors.black }}>
        <CardHeader
          titleTypographyProps={{ variant: 'h6' }}
          title={title}
          subheader={subheader}
          avatar={avatar}
        />
        {backgroundImage ? <div
          className={classes.cardImage}
          style={{ backgroundImage: `url('${backgroundImage}')` }}
        /> : undefined}
      </a>
      <CardContent>
        <Typography variant="body2" color="textSecondary" component="p">
          {body}
        </Typography>
        {children}
      </CardContent>
    </Card>
  );
}
ComponentCard.propTypes = {
  idx: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
  urlTarget: PropTypes.string,
  title: PropTypes.string.isRequired,
  subheader: PropTypes.string,
  avatar: PropTypes.element,
  backgroundImage: PropTypes.string,
  body: PropTypes.string
};