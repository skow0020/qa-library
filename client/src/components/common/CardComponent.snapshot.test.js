import CardComponent from './CardComponent';
import React from 'react';
import Chip from '@material-ui/core/Chip';
import renderer from 'react-test-renderer';
import Colors from 'utils/Colors';

describe('CardComponent Accessibility Tests', () => {
  test('CardComponent is accessible', async () => {
    const component = renderer.create(
      <CardComponent
        idx='article-card-7'
        url='www.snowflakes.com/wooo'
        urlTarget="_blank"
        title='Chirpy iguanas of guadalupe'
        subheader='By Corrin Tupple'
        avatar={<Chip
          size="small"
          label="UI Automation"
          style={{backgroundColor: Colors.blue}}
        />}
        backgroundImage="www.images.com/orcas.png"
        body="Well if there isn't somethin ado about nothing, then what?"
      />
    );
    
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
