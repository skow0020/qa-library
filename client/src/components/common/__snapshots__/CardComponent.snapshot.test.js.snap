// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardComponent Accessibility Tests CardComponent is accessible 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation1 MuiCard-root card-post MuiPaper-rounded"
  id="article-card-7"
>
  <a
    aria-label="Navigate to the url"
    href="www.snowflakes.com/wooo"
    rel="noopener noreferrer"
    style={
      Object {
        "color": "#000000",
      }
    }
    target="_blank"
  >
    <div
      className="MuiCardHeader-root"
    >
      <div
        className="MuiCardHeader-avatar"
      >
        <div
          className="MuiChip-root MuiChip-sizeSmall"
          onKeyUp={[Function]}
          style={
            Object {
              "backgroundColor": "#2196f3",
            }
          }
        >
          <span
            className="MuiChip-label MuiChip-labelSmall"
          >
            UI Automation
          </span>
        </div>
      </div>
      <div
        className="MuiCardHeader-content"
      >
        <span
          className="MuiTypography-root MuiCardHeader-title MuiTypography-h6 MuiTypography-displayBlock"
        >
          Chirpy iguanas of guadalupe
        </span>
        <span
          className="MuiTypography-root MuiCardHeader-subheader MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
        >
          By Corrin Tupple
        </span>
      </div>
    </div>
    <div
      className="makeStyles-cardImage-1"
      style={
        Object {
          "backgroundImage": "url('www.images.com/orcas.png')",
        }
      }
    />
  </a>
  <div
    className="MuiCardContent-root"
  >
    <p
      className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
    >
      Well if there isn't somethin ado about nothing, then what?
    </p>
  </div>
</div>
`;
